extensions/amp-base-carousel/1.0/scroller.js

```diff
@@ -160,68 +158,39 @@ function ScrollerWithRef(
       return;
     }
     setStyle(container, 'scrollBehavior', 'auto');
-    let position;
-    const slideWidth = container./* OK */ offsetWidth / visibleCount;
-    if (loop) {
-      if (snap) {
-        if (mixedLength) {
-          position = container.children[pivotIndex]./* OK */ offsetLeft;
-        } else {
-          position = slideWidth * pivotIndex;
-        }
-      } else {
-        if (mixedLength) {
-          position =
-            container.children[pivotIndex]./* OK */ offsetLeft +
-            scrollOffset.current;
-        } else {
-          position = mod(
-            scrollOffset.current + slideWidth * offsetRef.current,
-            container./* OK */ scrollWidth
-          );
-        }
-      }
-    } else {
-      if (snap) {
-        if (mixedLength) {
-          position = container.children[restingIndex]./* OK */ offsetLeft;
-        } else {
-          position = slideWidth * restingIndex;
-        }
-      } else {
-        if (mixedLength) {
-          position =
-            container.children[restingIndex]./* OK */ offsetLeft +
-            scrollOffset.current;
-        } else {
-          position = scrollOffset.current;
-        }
-      }
-    }
-    container./* OK */ scrollLeft = position;
+    ignoreProgrammaticScrollRef.current = true;
+    scrollContainerToElement(
+      axis,
+      alignment,
+      container,
+      container.children[pivotIndex],
+      scrollOffset.current
+    );
     setStyle(container, 'scrollBehavior', 'smooth');
-  }, [loop, mixedLength, restingIndex, pivotIndex, snap, visibleCount]);
+  }, [axis, alignment, loop, pivotIndex, restingIndex]);
 
   // Trigger render by setting the resting index to the current scroll state.
-  const debouncedResetScrollReferencePoint = useMemo(
-    () =>
-      debounce(
-        window,
-        () => {
-          // Check if the resting index we are centered around is the same as where
-          // we stopped scrolling. If so, we do not need to move anything.
-          if (
-            currentIndex.current === null ||
-            currentIndex.current === restingIndex
-          ) {
-            return;
-          }
-          setRestingIndex(currentIndex.current);
-        },
-        RESET_SCROLL_REFERENCE_POINT_WAIT_MS
-      ),
-    [restingIndex, setRestingIndex]
-  );
+  const debouncedResetScrollReferencePoint = useMemo(() => {
+    // Use local window if possible.
+    const win = containerRef.current
```

Comment:
One thing to confirm: the first render this always be global `window`. Is this ok?

Reply:
In that case, this will always be global `window` since the value from the first render will be memoized... I suppose we can put `debounce()` on a ref and update it in `useEffect` instead in case the window changes? :/

Reply:
That would make it pretty messy. I see that it will switch to local `win` as soon as the `restingIndex` changes? Perhaps it's good enough?

Reply:
Yes, that should be OK. Do we need to do anything to clean up the first instance?

Reply:
No. I think that's not very likely to be needed.